app/Domain/UserVacationStatsRetriever.php

```diff
@@ -103,6 +103,14 @@ public function getVacationDaysLimit(User $user, YearPeriod $yearPeriod): int
         return $limit ?? 0;
     }
 
+    public function hasVacationDaysLimit(User $user, YearPeriod $yearPeriod): bool
+    {
+        return $user->vacationLimits()
+            ->whereBelongsTo($yearPeriod)
+            ->first()
+            ?->hasVacation() ?? false;
```

Comment:
```suggestion
            ->first()?->hasVacation() ?? false;
```

This is more readable, I think.